internal/github/github.go

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+)
+
+const (
+	baseEndpoint = "https://api.github.com/repos"
+	timeout      = time.Second * 60
+)
+
+var client *http.Client
+
+func getClient() *http.Client {
+	if client == nil {
+		client = &http.Client{Timeout: timeout}
+	}
+
+	return client
+}
+
+func GetPullRequest(ctx context.Context, repository, pullRequestID string) ([]byte, error) {
+	endpoint := fmt.Sprintf("%s/%s/pulls/%s", baseEndpoint, repository, pullRequestID)
```

Comment:
I've used https://pkg.go.dev/github.com/google/go-github/v61 for a bunch of projects - will probably be better than doing type casting

Reply:
I was trying to avoid the dependency if we were only hitting a couple endpoints but given that this is almost certainly less brittle / better tested and isn't a completely random set of developers, I'll make the switch!

Reply:
go-github is intentionally pretty dependency light. https://github.com/google/go-github/blob/master/go.mod

I wouldn't worry about it too much.